src/resources/genre.ts

```diff
@@ -0,0 +1,206 @@
+import { Router, Request, Response } from 'express';
+import mongose from 'mongoose';
+
+import BadRequestError from '../errors/BadRequestError';
+
+import { Genres } from '../schemas/genres';
+
+export const getAllGenresRouter = Router();
+export const addNewGenreRouter = Router();
+export const updateGenreByTitleRouter = Router();
+export const deleteGenreByIdRouter = Router();
+
+const ObjectId = mongose.Types.ObjectId;
+
+/**
+ * @swagger
+ * /genres:
+ *   get:
+ *     summary: Get a list of all genres
+ *     description: Retrieve a list of all genres from the database.
+ *     responses:
+ *       200:
+ *         description: List of genres retrieved successfully.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: array
+ *               items:
+ *                 $ref: '#/components/schemas/Genre'
+ *       400:
+ *         description: Internal server error.
+ */
+getAllGenresRouter.get('/', async (req: Request, res: Response): Promise<void> => {
+    await Genres.find()
+    .then((genres) => {
+        if (genres.length === 0) {
+            throw new BadRequestError({code: 400, message: "No genres was found." });
+        }
+        res.status(200).json(genres);
+    })
+    .catch((err: Error) => {
+        throw new BadRequestError({code: 400, message: err.message });
+    });
+  });
+  
+  /**
+   * @swagger
+   * /genres/{name}:
+   *   post:
+   *     summary: Add a new genre
+   *     description: Add a new genre to the database.
+   *     parameters:
+   *       - in: path
+   *         name: name
+   *         required: true
+   *         description: The name of the genre.
+   *         schema:
+   *           type: string
+   *     requestBody:
+   *       description: Genre object to add.
+   *       required: true
+   *       content:
+   *         application/json:
+   *           schema:
+   *             type: object
+   *             properties:
+   *               name:
+   *                 type: string
+   *                 example: Action
+   *     responses:
+   *       201:
+   *         description: Genre added successfully.
+   *       400:
+   *        description: Bad request - Genre not provided.
+   *       404:
+   *         description: Bad request - Invalid data provided.
+   */
+  
+  addNewGenreRouter.post('/:name', async (req: Request, res: Response): Promise<void> => {
+    const { name } = req.params;
+
+    if (!name) {
+        throw new BadRequestError({code: 400, message: "Genre not provided." });
+    }
```

Comment:
You can improve input data by adding complex validation with Joi package.

Example: https://www.digitalocean.com/community/tutorials/how-to-use-joi-for-node-api-schema-validation

Reply:
done